glue-codes/openfast-cpp/src/FAST_Prog.cpp

```diff
@@ -11,23 +11,61 @@ inline bool checkFileExists(const std::string& name) {
 void readTurbineData(int iTurb, fast::fastInputs & fi, YAML::Node turbNode) {
 
   //Read turbine data for a given turbine using the YAML node
-  fi.globTurbineData[iTurb].TurbID = turbNode["turb_id"].as<int>();
-  fi.globTurbineData[iTurb].FASTInputFileName = turbNode["FAST_input_filename"].as<std::string>() ;
-  fi.globTurbineData[iTurb].FASTRestartFileName = turbNode["restart_filename"].as<std::string>() ;
+  if (turbNode["turb_id"]) {
```

Comment:
Recommend building some higher level YAML utilities and using it here.

Reply:
Agreed. Implemented in 9f34f9eb119d38dc0c78bfe58fdd23d084d02430. Reduced the code quite a bit.